### PR TITLE
Ported to linter-plus.

### PR DESCRIPTION
**Pull request by keplersj**
Merge when AtomLinter/linter-plus#24 closes.

### CODE REVIEW

File: lib/linter-crystal.coffee [hunk 1]
```diff
@@ -1,46 +1,56 @@
...
+    lint: @lint
+    lintOnFly: false
+  }
 
-  destroy: ->
-    @listen.dispose()
+  lint: (TextEditor) ->
+    regex = /.+ in (.+):(\d+): (.+)/
+    return new Promise (Resolve) ->
+      if TextEditor.getPath()
```
Comment (steelbrain) [inline lib/linter-crystal.coffee]: this line is redundant https://github.com/AtomLinter/linter-ruby/pull/18/files#r33106620

**Comment by keplersj** [comment]
I'm considering deprecating this package and replacing it with my Crystal IDE package.

**Comment by steelbrain** [comment]
Instead of just leaveing it as is, You might want to remove all the codebase and add an error notification on startup so user would know what to install and leave it as is.

**Comment by keplersj** [comment]
Fantastic idea! I'll get on that once I clean out my notifications.

**Pull request merged**